### PR TITLE
insertAt support for array

### PR DESCRIPTION
**Pull request by abchatra**
This adds insertAt support for array. Also updates removeAt to return the removed element and shift the rest of the elements.

### CODE REVIEW

File: libs/core/pxt.cpp [hunk 1]
```diff
@@ -269,17 +288,63 @@ namespace pxt {
       return 0;
     }
 
-    void Segment::remove(uint32_t i) 
+    //this function removes index i and
```
Comment (riknoll) [inline libs/core/pxt.cpp]: Comment seems incomplete here

File: libs/core/pxt.cpp [hunk 2]
```diff
@@ -269,17 +288,63 @@ namespace pxt {
...
+        this->print();
+#endif  
+        return ret;
       }
-      return;
+      error(ERR_OUT_OF_BOUNDS);
+      return 0;
+    }
+
+    //this function removes index i and
```
Comment (riknoll) [inline libs/core/pxt.cpp]: I think this is a copy-pasted duplicate of the comment on remove()

**Comment by riknoll** [comment]
@abchatra build is failing right now

**Comment by abchatra** [comment]
Fixed review comments

**Pull request merged**